### ISSUE TITLE
can't send messages anymore

### ISSUE THREAD

**Issue by jaymzh** [dequis/purple-facebook#549]
I can no longer send messages, doing so kicks me off. Receiving messages works fine (including my own that I send via web/mobile/etc.). Anyone else?

**Comment by jaymzh** [comment]
Oh I should add it just says "Failed to send message" - and that's it.

**Comment by ryanford** [comment]
Also experiencing this. Buddy list loads as normal but attempting to send message will error and disconnect with "Failed to send message" as only error message. I think it's been going several days prior to this issue opening.

**Comment by advcomp2019** [comment]
I believe this is a part of the end-to-end encryption. I have this issue with people that I know is using end-to-end encryption, but I do not have this if they have not this enabled.

**Comment by jaymzh** [comment]
How can you tell who this is enabled for? Before, when it was "secret conversations", it was a whole separate thread, but now that it's rolling out as a default... how do you know which people it's enabled for? I can't find anything about it in the 'info' thing for various chats.

**Comment by advcomp2019** [comment]
As far as I know, if you are on the Android Messenger app, you will see a message talking about end-to-end encryption. Beyond that, I do not know.

**Comment by Metalhead33** [comment]
This is caused by #547 - until someone updates the plugin for that, you're out of luck, buddy.

**Comment by jaymzh** [comment]
I don't think so, because I still receive all messages. When E2E is activated ... You can't receive the messages either, since they're encrypted (as described in the issue you linked).

**Comment by advcomp2019** [comment]
As far as I can tell with mine, when you see the second photo on this Reddit post: https://www.reddit.com/r/facebook/comments/1bn8ojj/help_messenger_forced_me_to_have_endtoend/ , I can not send or receive messages to the person, and I see the same issue as yours when I send.

If the person does not have that, I can send a message to that person from Pidgin just fine.

**Comment by advcomp2019** [comment]
I just found another way to check for end-to-end encryption if you did not see that message.

If you on the web browser, the URL with e2ee is "https://www.facebook.com/messages/e2ee/t/<FacebookID>" While users without end-to-end, the URL is "https://www.facebook.com/messages/t/<FacebookID>"

**Comment by jaymzh** [comment]
Thanks. Yeah that checks out. None of my regular chats are e2ee yet.

Which makes sense, if they were, Pidgin wouldn't be able to read both sides of the messages.

That said, it may be related. For example, there may be some new "Check is this is supposed to be encrypted" call required to be called first that then gives you a context you pass when you send messages. Who knows.

**Comment by debmanlinux** [comment]
As of May 5th, I am having exactly the same problem. Pidgin starts & shows Facebook Friends. However when I try to send a message I get "Disconnected, failed to send message". After a short time, it reconnects, but the same thing happens when I try to send again. Even when end-to-end encryption was not enabled, I had the problem that Pidgin would not send.

**Comment by swinokur** [comment]
Hey all, 

I am seeing the same issue as others, but also thought that I'd mention (as a potential pathway towards a plugin implementation) that Beeper (https://github.com/beeper) is working fine with Facebook e2e messages. (people I can't message in Pidgin, I can message in beeper).

this (https://github.com/mautrix/meta) appears to be the "bridge" that beeper uses to talk to meta.

**Comment by fonzi** [comment]
If it's just matrix, making beeper work, it might be better to just fork this https://github.com/matrix-org/purple-matrix and point to a local messenger bridge. It would also pull Instagram messages.

**Comment by maccagnan** [comment]
Same issue here, on Facebook I see contacts and receive messages fine, but can't send messages out.

**Comment by joonas-foo** [comment]
I alao get this error, "joonas.hoglund disconnected - Failed to send message". Could someone else update the plugin if the author doesn't have time for that?!
